functions/normalize_php_ext.pp

```diff
@@ -48,36 +48,27 @@ function php::normalize_php_ext(String $php_extension_name, Optional[String] $ph
 	]
 
 	$real_php_version = $php_version ? {
-		undef => pick(lookup('profiles::web::php_version'), $facts['php_version'][0,3]),
+		undef => pick(lookup('php::params::php_version'), $facts['php_version'][0,3]),
 		default => $php_version
 	}
 
-	# for this to work, this function MUST be placed somewhere AFTER the respective resource declaration
-	if (defined(Apt::Ppa['ppa:ondrej/php5-oldstable']) or
-		  defined(Apt::Ppa['ppa:ondrej/php5']) or
-		  defined(Apt::Ppa['ppa:ondrej/php5-5.6']) or
-	    defined(Apt::Ppa['ppa:ondrej/php'])) or
-		  ####################################
-		  defined(Apt::Source['netlogix-php']) or
-		  defined(Apt::Source['netlogix-php70']) or
-			defined(Apt::Source['php5-oldstable']) {
-		$custom_source_defined = true
-	} else {
-		$custom_source_defined = false
+	$sources_filtered = $sources.filter |$source| {
+		defined(Apt::Source[$source]) or defined(Apt::Ppa[$source])
 	}
+	$custom_source_defined = length($sources_filtered) > 0
 
-	$real_custom_source = $custom_source ? {
+	$has_custom_source = $custom_source ? {
 		true => true,
 		default => $custom_source_defined
 	}
 
-	if ($real_custom_source) {
-		$real_package_name = "php${real_php_version}-${php_extension_name}"
+	if (!$has_custom_source and member($plain_package_names, "php-${php_extension_name}")) {
+		$real_package_name = "php-${php_extension_name}"
 	} else {
-		if (member($plain_package_names, "php-${php_extension_name}")) {
-			$real_package_name = "php-${php_extension_name}"
-		} else {
-			$real_package_name = "php${real_php_version}-${php_extension_name}"
+		$real_package_name = "php${real_php_version}-${php_extension_name}"
+		# resources work in puppet > 4 inside functions
+		package { "php-${php_extension_name}":
```

Comment:
This gives a function that is used for generating a package name a huge side effect. And what happens if the function is called twice for the same extension?